vertical-pod-autoscaler/pkg/admission-controller/logic/server.go

```diff
@@ -127,23 +127,31 @@ func (s *AdmissionServer) getContainerPatch(pod v1.Pod, i int, patchKind string,
 		pod.Spec.Containers[i].Resources.Requests == nil {
 		patches = append(patches, getPatchInitializingEmptyResources(i))
 	}
-	// Add empty resources requests object if missing
-	if pod.Spec.Containers[i].Resources.Requests == nil {
-		patches = append(patches, getPatchInitializingEmptyResourcesSubfield(i, "requests"))
-	}
+
 	annotations, found := annotationsPerContainer[pod.Spec.Containers[i].Name]
 	if !found {
 		annotations = make([]string, 0)
 	}
-	for resource, request := range containerResources.Requests {
-		// Set request
-		patches = append(patches, getAddResourceRequirementValuePatch(i, "requests", resource, request))
-		annotations = append(annotations, fmt.Sprintf("%s request", resource))
-	}
+
+	patches, annotations = appendPatchesAndAnnotations(patches, annotations, pod.Spec.Containers[i].Resources.Requests, i, containerResources.Requests, "requests", "request")
+	patches, annotations = appendPatchesAndAnnotations(patches, annotations, pod.Spec.Containers[i].Resources.Limits, i, containerResources.Limits, "limits", "limit")
+
 	updatesAnnotation := fmt.Sprintf("container %d: ", i) + strings.Join(annotations, ", ")
 	return patches, updatesAnnotation
 }
 
+func appendPatchesAndAnnotations(patches []patchRecord, annotations []string, current v1.ResourceList, containerIndex int, resources v1.ResourceList, fieldName, resourceName string) ([]patchRecord, []string) {
+	// Add empty requests object if it's missing and we're about to fill it.
```

Comment:
What I meant is that this is not necessarily a request object any more, this can be either request or limit.

Reply:
Done?